File: TwitterPage/Common/Services/ParseFieldChecker.m
//
//  ParseFieldChecker.m
//  TwitterPage
//
//  Created by DMITRY SINYOV on 30.08.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

#import "ParseFieldChecker.h"

@implementation ParseFieldChecker

+(NSString*)checkStringOrDecimalNumberToString:(id)checkedObject
{
    if ([checkedObject isKindOfClass:[NSString class]])
    {
        return checkedObject;
    }
    else if ([checkedObject isKindOfClass:[NSNumber class]])
    {
        if ([checkedObject isEqualToNumber:@(((NSNumber*)checkedObject).longLongValue)])
        {
            return [NSString stringWithFormat:@"%lld", ((NSNumber*)checkedObject).longLongValue];
        }
        else return nil;
    }
    else return nil;
}
+(NSNumber*)checkStringOrDecimalNumberToNumber:(id)checkedObject
{
    if ([checkedObject isKindOfClass:[NSNumber class]])
    {
        if ([checkedObject isEqualToNumber:@(((NSNumber*)checkedObject).longLongValue)])
        {
            return checkedObject;
        }
        else return nil;
    }
    else if ([checkedObject isKindOfClass:[NSString class]])
    {
        return @([(NSString*)checkedObject longLongValue]);
    }
    else return nil;
}

+(id)fieldDataForJSONfield:(id)field
{
    if (field)
    {
        return [[self class] fixNull:field];
    }
    else return nil;
}

+(id)fixNull:(id)someObject
{
    if ([someObject isKindOfClass:[NSNull class]])
    {
        return nil;
    }
    return someObject;
}

@end

File: SWIFT/TwitterPageSWIFT/Common/Services/ParseFieldChecker.swift
//
//  ParseFieldChecker.swift
//  TwitterPageSWIFT
//
//  Created by DMITRY SINYOV on 15.09.17.
//  Copyright © 2017 DMITRY SINYOV. All rights reserved.
//

import UIKit

class ParseFieldChecker: NSObject {
    class func fixNull<T>(object:T?)->T? {
        if (object is NSNull) { return nil }
        return object
    }
    class func fieldDataForJSON<T>(field: T?)->T? {
        if field != nil {
            return self.fixNull(object: field)
        }
        return nil
    }
    class func checkStringOrDecimalNumberToString(checkedObject: AnyObject?) -> String? {
        if (checkedObject is String) { return checkedObject as? String}
        else if (checkedObject is NSNumber) {
            return String(format:"%lld", (checkedObject as! NSNumber).int64Value)
        }
        else { return nil }
    }
    class func checkStringOrDecimalNumberToNumber(checkedObject: AnyObject?) -> NSNumber? {
        if (checkedObject is NSNumber) { return checkedObject as? NSNumber}
        else if (checkedObject is String) {
            return NSNumber(value:Int((checkedObject as! String))!)
        }
        else { return nil }
    }
}
